### PR TITLE
Basic mentions

### PR DESCRIPTION
**Pull request by bobbybee**
* Refactor mention conversion code
* Bolden display of mentions
* Converts tab-completable IRC style mentions to Discord mentions
* Support IRC style mentions by local aliases as well (**slow** for large servers; should be cached!)

### CODE REVIEW

File: libdiscord.c [hunk 1]
```diff
@@ -1423,6 +1425,101 @@ discord_replace_emoji(const GMatchInfo *match, GString *result, gpointer user_da
...
+
+		while (g_hash_table_iter_next (&iter, &key, &value)) {
+			/* Key is the user ID, value is DiscordUser */
+
+			DiscordUser *u = value;
+			gchar *username = discord_create_fullname(u);
+			PurpleBuddy *buddy = purple_blist_find_buddy(da->account, username);
+			g_free(username);
+
+			if(buddy && g_strcmp0(buddy->alias, identifier) == 0) {
```
Comment (EionRobb) [inline libdiscord.c]: use `purple_strequal` instead of `g_strcmp0 == 0` for readability

File: libdiscord.c [hunk 2]
```diff
@@ -1423,6 +1425,101 @@ discord_replace_emoji(const GMatchInfo *match, GString *result, gpointer user_da
...
+
+	return FALSE;
+}
+
+static gchar *
+discord_replace_mentions_bare(DiscordAccount *da, gchar *message)
+{
+	gchar *tmp = g_regex_replace_eval(mention_regex, message, -1, 0, 0, discord_replace_mention, da, NULL);
+	return tmp != NULL ? tmp : message;
+}
```
Comment (EionRobb) [inline libdiscord.c]: What's supposed to happen here, memory-wise?  Should the callback be `g_free()`d or is it an in-place modification?

**Comment by EionRobb** [comment]
Good enough for a choccy fish

**Comment by thundergreen** [comment]
I got th other way around problem. If I am mentioned I dont get those messages. ANy ideas?

**Pull request merged**